packages/medusa/src/types/routing.ts

```diff
@@ -1,14 +1,19 @@
+import type { Customer, User } from "../models"
 import type { NextFunction, Request, Response } from "express"
 
-import type { Customer, User } from "../models"
 import type { MedusaContainer } from "./global"
 
 export interface MedusaRequest extends Request {
   user?: (User | Customer) & { customer_id?: string; userId?: string }
   scope: MedusaContainer
   session?: any
   requestId?: string
-  auth_user?: { id: string; app_metadata: Record<string, any>; scope: string }
+  auth?: {
+    actor_id: string
+    auth_user_id: string
```

Comment:
Wouldn't this and the actor_id point to the same ID?

Reply:
The `auth_user_id` would be the ID of the authenticated user from the Auth module, while as the `actor_id` will be the ID of the entity the authenticated user is possibly associated with.

Example: User <> Auth
```
auth_user_id: "authusr_1234"
actor_id: "usr_1234"
```

Example: Customer <> Auth
```
auth_user_id: "authusr_1234"
actor_id: "cus_1234"
```

Example: Secret API key
```
auth_user_id: undefined
actor_id: "sk_1234"
```

Reply:
@olivermrbl ok understood the difference, but the naming is still confusing in my opinion. For example, tokens and users should be treated equally (they are both actors, as an actor is the one performing the operation), but an entity (eg. customer, or store) would be something completely different. Anyway it's not a blocker, but I think we need to rethink our naming and approach here.

Reply:
In the examples above, we are treating the users and tokens equally wrt the actor. Auth user is a separate concept from entities Customer or User and is specifically designed to provide authentication to those. Do you think the auth user should be renamed? I am not sure I understand what is confusing here.

Reply:
@olivermrbl Ok I think the part that got me confused is what `customer` means, I thought it represents an `organization`, but now I remembered that it's simply a `storefront user` basically. All clear now 👍